src/TwigComponent/src/ComponentFactory.php

```diff
@@ -36,38 +36,34 @@ public function __construct(ServiceLocator $components, PropertyAccessorInterfac
     }
 
     /**
-     * @param string|object $component Component name as string or component object
+     * @param string|object $name Component name, component object or component FQCN
      */
-    public function configFor($component, string $name = null): array
+    public function configFor(string|object $name): array
```

Comment:
What about returning some new class - e.g. `ComponentConfig`?

This would make typing on the `PreRenderEvent` nicer. But it's also motivated by an urge to (as a sanity check) make sure that if a user needs to "do crazy things manually", their code looks sane.

I'm looking and thinking about what it would look like if someone wanted to render a component manually... and maybe also pass their own custom custom. For example:

```php
$component = $componentFactory->create('alert', ['message' => 'foo']);
$config = new ComponentConfig('components/alert.html.twig', AlertComponent::class, isLive: true);

echo $componentRenderer->render($component, $config);
```

Reply:
Yes, that's on my radar!

Reply:
Are you thinking just a simple object?

```php
class ComponentConfig
{
    public function __construct(private array $config) {}

    public function get(string $key, mixed $default = null): mixed {}
}
```

Reply:
Yup, simple object, exactly :). I would prefer to have as many named args in the constructor as we can, before having the final "everything else" `$config`, but no string opinion. The specific args just make the class easier to understand (e.g. `$template`).

`ComponentMetadata` sounds excellent.